File: callback/callback.go
// Copyright © 2015 FlexibleBroadband Team.
// All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.
//	      ___ _           _ _     _
//	     / __\ | _____  _(_) |__ | | ___
//	    / _\ | |/ _ \ \/ / | '_ \| |/ _ \
//	   / /   | |  __/>  <| | |_) | |  __/
//	   \/    |_|\___/_/\_\_|_.__/|_|\___|

package callback

import (
	"net/http"
	"net/url"
)

// App
type AppMapping interface {
	AddrMapping(public string) (private string)
}

// RTMP
type RTMP interface {
	RtmpAccessCheck(remote, local, appname, path string) bool
}

// HTTP-FLV
type FLV interface {
	FlvAccessCheck(remote, url, path string, form url.Values, cookies []*http.Cookie) bool
}

// IM
type IM interface {
	IMAccessCheck(remote, url, path string, form url.Values, cookies []*http.Cookie) (uname string, access byte, ok bool)
}

type FlvCallback interface {
	FLV
}

type IMCallback interface {
	IM
}

type RTMPCallback interface {
	AppMapping
	RTMP
}

File: test/lua/callback.lua
-- file's path: $GOPATH/src/github.com/Alienero/IamServer/test/lua/

-- module 
callback = {}

-- var
callback.mapping = {[""]="master",["master"]="master"}

-- callback functions

function callback.addr_mapping(public)
	return callback.mapping[public]
end

function callback.rtmp_access_check(remote, local_addr, appname, path)
	print("remote:",remote)
	print("local:",local_addr)
	print("appname:",appname)
	print("path:",path)

	if appname=="live" and path == "master" then
		return true
	else
		return false
	end
end

function callback.flv_access_check(remote, url, path,froms, cookies)
	print("remote",remote)
	print("url:",url)
	print("path:",path)

	if path == "/live/master" then
		return true
	else
		return false
	end
end

function callback.im_access_check(remote, url, path,froms, cookies)
	print("remote",remote)
	print("url:",url)
	print("path:",path)

	if path == "/im/master" then
		return "I am master",1,true
	else
		return "",0,false
	end
end

return callback